### DISCUSSION TITLE
Accessing opencrx/openmdx database via jdbc

### DISCUSSION THREAD

**Discussion by nyemiks** [openmdx/openmdx#69] [Q&A]
For some rare situations, I see the need to be able to access the data from an opencrx database via jdbc. Are there any examples of this ? if yes please share. How do I ensure that I fetch data from only one segment ? I am also interested in seeing what an SQL update query will look like if possible.

A rare situation could be to allow a third party app pull information from the database using jdbc. In such a scenario, I will want to write an sql query and provide the sql query to the third party app vendor to use in his app to pull the data. In such a scenario exposing the data as web service is not an option due to limitations from the third party vendor app. (I know this is rare but situations like this do occur)

Another scenario could be for mapping applications.

I create spatial geometry column for an opencrx entity e.g. activity, or warehouse then create a view that exposes this geometry column along side other fields to a map server application like GeoServer (https://geoserver.org/). The geometry column will be managed via jdbc using spatial java api's e.g geotools (https://www.geotools.org/). I know openmdx does not support spatial data types so I see jdbc as a good work around to add spatial capabilities to the rich opencrx database. My jdbc calls can be wrapped inside an openmdx operation.

**Comment by DlRTY-HARRY** [comment]
The SQL query and update statements are logged at detail level for you to be inspected:
- openMDX acquires the DataSource via JNDI lookup and so may your application
- Enable [openMDX query logging](https://github.com/openmdx/openmdx/blob/3e281976aa4afd5ec04a81d5f26072231f764804/core/src/main/java/org/openmdx/base/dataprovider/layer/persistence/jdbc/Database_2.java#L1946) to see, how your read requests shall be transformed into SQL statements
- Enable [openMDX update logging](https://github.com/openmdx/openmdx/blob/3e281976aa4afd5ec04a81d5f26072231f764804/core/src/main/java/org/openmdx/base/dataprovider/layer/persistence/jdbc/Database_2.java#L1964) to see, how your write requests shall be transformed into SQL statements
